docs/2025/puzzles/day05.md

```diff
@@ -2,10 +2,245 @@ import Solver from "../../../../../website/src/components/Solver.js"
+We'll need to add a couple methods to `LRange` to handle intersection and difference:
+
+```scala
+final case class LRange(start: Long, end: Long):
+```
+For intersection, there are more or less 4 cases:
+* Overlaps on the left edge
+* Overlaps on the right edge
+* Overlaps internally
+* Overlaps entirely
+
+Let's walk through each case.
+
+When we overlap on the left edge, it will look something like this:
+```
+   *--------* - Us
+*-------*     - Them
+```
+
+* Our start >= their start, <= their end
+* Our end >= their start, >= their end
+
+When we overlap on the right edge, it will look something like this:
+```
+*-------*     - Us
+   *--------* - Them
+```
+
+* Our start <= their start, <= their end
+* Our end >= their start, <= their end
+
+When we overlap interally, it will look something like this:
+
+```
+*--------------* - Us
+    *-------*    - Them
+```
+
+* Our start <= their start, <= their end
+* Our end >= their start, >= their end
+
+When we are entirely overlapped, it will look something like this:
+
+```
+    *-------*    - Us
+*--------------* - Them
+```
+
+* Our start >= their start, <= their end
+* Our end >= their start, <= their end
+
+In all cases, our start is <= their end, and our end is >= their start.
+We can also observe that to get the overlap, we take the maximum start, and the minimum end.
+
+So let's add that intersect method:
+
+```scala
+  infix def intersect
+    (
+      t: LRange
+    ): Option[LRange] =
+    Option.when(end >= t.start && start <= t.end):
+      LRange(start max t.start, end min t.end)
+```
+
+We needed that intersect to implement difference. The intersect will clamp any points that go outside our bound to 
+our edge, meaning we can compare easier.
+
+We can do simple math to make the set - we find the points before and after the hole. If we didn't intersect at all,
+then we can just return a singleton set with `this`.
+
+```scala
+  infix def -
+    (
+      that: LRange
+    ): Set[LRange] =
+    this intersect that match
+      case Some(hole) =>
+        var daSet = Set.empty[LRange]
+        // if start == hole.start then there won't be any points left before the hole starts
+        if start != hole.start then daSet += LRange(start, hole.start - 1)
+        // if end == hole.end then there won't be any points left after the hole ends
+        if end != hole.end then daSet += LRange(hole.end + 1, end)
+        daSet
+      case _ => Set(this)
+```
+
+Now, for part 2, we'll just need to iteratively combine all these ranges.
+
+```scala
+def part2(input: String): Long =
+  val (ranges, _) = parse(input)
+  val combinedRanges =
+    ranges.foldLeft(Set.empty[LRange]): (acc, range) =>
+      // remove the new range from everything first to prevent overlaps
+      val removed = acc.flatMap(_ - range)
+      // then add it seperately
+      removed + range
+  // toIterator to prevent Set from deduplicating our result
+  combinedRanges.toIterator.map(_.size).sum
```

Comment:
Thanks for the writeup, I think it's written well! Minor nitpick: I believe `.toIterator` has been deprecated since 2.13, and should be replaced with `.iterator` instead.